Formula/percona-xtrabackup.rb

```diff
@@ -0,0 +1,56 @@
+
+class PerconaXtrabackup < Formula
+  desc "Open source hot backup tool for InnoDB and XtraDB databases"
+  homepage "https://www.percona.com/software/mysql-database/percona-xtrabackup"
+  url "https://www.percona.com/downloads/XtraBackup/Percona-XtraBackup-2.4.2/source/tarball/percona-xtrabackup-2.4.2.tar.gz"
+  sha256 "faeac6f1db4a1270e5263e48c8a94cc5c81c772fdea36879d1be18dcbcd1926e"
+
+  option "with-man-pages", "Build man pages. Needs python-sphinx to be installed"
+
+  depends_on "cmake" => :build
+  depends_on "libev" => :build
+  depends_on "libgcrypt"
+  depends_on :mysql => :recommended
+
+  def install
+    cmake_args = %W[
+      -DBUILD_CONFIG=xtrabackup_release
+      -DDOWNLOAD_BOOST=1
```

Comment:
Percona Xtrabackup actually builds a custom version of MySQL 5.7, which requires Boost 1.59.  I first tried declaring the dependency directly as `depends_on "boost" => :build` but the build failed because homebrew's boost is v1.60.

I noticed this commit in a legacy mysql formula: https://github.com/Homebrew/legacy-homebrew/commit/faef69a4167870f994b48d750dad410094f86b85

Even though the approach used in that commit works for this formula, it's really similar to letting the build download it, but with the disadvantage that the version is hardcoded here instead of being automatically resolved.  In that sense, I'd think it's better to leave the downloading to the build.

The current version of the MySQL formula uses a source distribution bundled with boost.  Sadly, there doesn't seem to be a similarly bundled version of percona-xtrabackup.

Which approach would you recommend here?